src/composition.rs

```diff
@@ -551,11 +578,32 @@ impl Composition {
                 imports: vec![],
                 eof: vec![],
                 body: vec![ast::Statement::Expr(Box::new(
-                    expr_statement,
+                    expr_statement.to_owned(),
                 ))],
             };
-            match flux::formatter::convert_to_string(&file) {
-                Ok(text) => return Some(text),
+            match (
+                flux::formatter::convert_to_string(&file),
+                &expr_statement.expression,
```

Comment:
Yeah, this is what I was wondering. This doesn't really communicate the purpose and intent of the code as well as we probably want. We're combining two orthogonal concerns (formatted string and position information), and we're only really matching on one thing (I'm surprised `clippy` didn't complain).

Reply:
See if the updated method naming and signature, addresses your concern.